File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // Brute Force
        // for (int i=0; i<nums.size(); i++) {
        //     for (int j=0; j<nums.size(); j++) {
        //         if (i == j) 
        //             continue;
        //         if ((nums[i] + nums[j]) == target)
        //             return {i, j};
        //     }
        // }
        // return {};
        
        //Map
        unordered_map<int, int> map;
        for (int i=0; i<nums.size(); i++) {
            int diff = target - nums[i];
            if (map.find(diff) != map.end()) {
                return {map[diff], i};
            }
            map[nums[i]] = i;
        }

        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # res = []
        # for i, x in enumerate(nums):
        #     check = set([y for j,y in enumerate(nums) if j !=i ])
        #     if target - x in check:
        #         nums[i] = True
        #         t = nums.index(target - x)
        #         return [i, t]
        # return []

        check = {}
        for i, x in enumerate(nums):
            if target - x in check:
                return [i, check[target-x]]
            check[x] = i
        return []